drivers/arm/i2c_master.c

```diff
@@ -32,6 +32,15 @@
 
 static uint8_t i2c_address;
 
+// ChibiOS uses two initialization structure for v1 and v2/v3 i2c APIs.
```

Comment:
@yiancar since I think you're more familiar with this, any issues with this?

Reply:
Is there anything else I should improve about this? Or is this good to merge?

Reply:
I actually have never checked what API the f1 uses. Just to make sure @awkannan any takes on this?

Reply:
I'd prefer more general means of determining the i2c code to use rather than a specific MCU symbol definition, although ChibiOS doesn't exactly make it easy. Even if it was just the generic family definition instead (`STM32F1xx`) I'd be happy enough.

Reply:
I haven't tested this i2c code on STM32F1xx.

I agree with @pelrun. There are also multiple i2c buses available to use on STM chips, so this still is not enough. 

Until we figure out a good way to handle i2c bus selection and using family definition, I'd personally prefer to keep this in keyboard branches, etc. Just override i2c_master in your local keyboard folder. But I'm not the judge of that, I defer to QMK people to figure it out. The reason I say this though is because my RGB code is stuck in purgatory because it doesn't support every STM32 chip yet.

Also, as it stands, any non STM32F1xx chip would fall back to the F303 code, which is wrong on many chips. I like how we do it in eeprom, where we raise an error if no one has written the code for a specific family.

I also don't know how I feel about making tons of ifdefs, but I don't know a better alternative.

Reply:
The best solution would probably be to add an explicit symbol in the board definition and choose based on that. I was planning on bringing in to core the main board definitions that are floating around some of the keyboards, at which point it's fairly straightforward to keep them in line.

Reply:
I agree with @pelrun, however I didn't make it so that you have to declare this in board.h because I feel like that would break compatibility with a lot of existing code. I am thinking about either let the board.h define the config structure and initialization, thus they can both control the i2c version and which i2c bus to use. However, again, this would come at the cost of backwards compatibility. I initially just provided my own i2c_master file, but @drashna suggested to me that I should probably use the standard i2c_master. So I modified the i2c master.

Reply:
If this is an issue right now, then it should be fine to revert back to the separate files.  

Though, I'm not fond of having multiple files, but it seems like this is the best option until we have a better way to handle this.